### PR TITLE
Fix linux process start_time

### PR DESCRIPTION
**Pull request by jnasselle**
|Related issue|
|---|
|#11114|

## Description

Hi team!

This PR will tweak @LubinLew contribution PR #11227 according to Wazuh SDLC.
Changes include fixing sysinfo processes `start_time` value.

## Tasks
- Change `linuxProcessHelper.h` filename since their functions could be useful for other implementations no related to processes.
- Add EOL in every file
- Avoid `#ifndef WIN32` by adding this files to linux-specific targets in CMakeFiles.txt
- Fix manager-side implementation when saving start_time, specifically retrieving the value from cJSON struct. This should be obtained from `valuedouble` member variable
- Change strol to stroll to guarantee minimum sizes.
## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
  - [ ] Windows
  - [ ] MAC OS X
- [x] Source installation
- [ ] Package installation
- [ ] Source upgrade
- [ ] Package upgrade
- [ ] Review logs syntax and correct language
- [ ] QA templates contemplate the added capabilities

- Memory tests for Linux
  - [x] Scan-build report
  - [ ] Coverity
  - [x] Valgrind (memcheck and descriptor leaks check)
  - [ ] Dr. Memory
  - [ ] AddressSanitizer
- Memory tests for Windows
  - [ ] Scan-build report
  - [ ] Coverity
  - [ ] Dr. Memory
- Memory tests for macOS
  - [ ] Scan-build report
  - [ ] Leaks
  - [ ] AddressSanitizer

- [x] Retrocompatibility with older Wazuh versions
- [ ] Working on cluster environments
- [ ] Configuration on demand reports new parameters
- [ ] The data flow works as expected (agent-manager-api-app)
- [ ] Added unit tests (for new features)
- [ ] Stress test for affected components

- Decoder/Rule tests
  - [ ] Added unit testing files ".ini"
  - [ ] runtests.py executed without errors

### CODE REVIEW

File: src/wazuh_db/wdb.h [hunk 1]
```diff
@@ -715,10 +715,10 @@ int wdb_package_update(wdb_t * wdb, const char * scan_id);
...
 
 // Insert process info tuple. Return 0 on success or -1 on error.
-int wdb_process_insert(wdb_t * wdb, const char * scan_id, const char * scan_time, int pid, const char * name, const char * state, int ppid, int utime, int stime, const char * cmd, const char * argvs, const char * euser, const char * ruser, const char * suser, const char * egroup, const char * rgroup, const char * sgroup, const char * fgroup, int priority, int nice, int size, int vm_size, int resident, int share, int start_time, int pgrp, int session, int nlwp, int tgid, int tty, int processor, const char * checksum, const bool replace);
+int wdb_process_insert(wdb_t * wdb, const char * scan_id, const char * scan_time, int pid, const char * name, const char * state, int ppid, int utime, int stime, const char * cmd, const char * argvs, const char * euser, const char * ruser, const char * suser, const char * egroup, const char * rgroup, const char * sgroup, const char * fgroup, int priority, int nice, int size, int vm_size, int resident, int share, long long start_time, int pgrp, int session, int nlwp, int tgid, int tty, int processor, const char * checksum, const bool replace);
```
Comment (HanesSciarrone) [inline src/wazuh_db/wdb.h:718]: Hi @jnasselle 

Is there a possibility that the `start_time` field grows large enough to produce an erroneous read if it is signed? In fact, the conversion used in the linuxInfoHelper.h file was from string to unsigned long long.

Best regards
Hanes

Comment (jnasselle) [reply src/wazuh_db/wdb.h:718]: Hi @HanesSciarrone ,

You are right, but sqlite3 db does not support storing unsigned integers. That's why the helper retrieve `uint64_t` but is statically casted to `int64_t` when this data is returned by sysinfo (this data will be sync to local agent db using signed integer).

Regards,
Nico

**Review by HanesSciarrone — DISMISSED** [review]

File: src/shared_modules/utils/linuxInfoHelper.h [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            if (0UL == btime)
+            {
+                const std::string key {"btime "};
+                const auto file { Utils::getFileContent("/proc/stat") };
+
+                btime = std::stoull(file.substr(file.find(key) + key.length()));
+            }
+        }
+        // LCOV_EXCL_START
+        catch (...)
```
Comment (RamiroRD) [inline src/shared_modules/utils/linuxInfoHelper.h:41]: Wouldn't we want to log a warning/error or something?

File: src/shared_modules/utils/linuxInfoHelper.h [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        {}
+
+        // LCOV_EXCL_STOP
+
+        return btime;
+    }
+
+    static uint64_t getClockTick(void)
```
Comment (RamiroRD) [inline src/shared_modules/utils/linuxInfoHelper.h:49]: For better readability, we could add comments describing what the value returned is, what physical dimensions it is in, etc.

**Review by RamiroRD — CHANGES_REQUESTED** [review]

File: src/shared_modules/utils/linuxInfoHelper.h (hunk 2 above)
Comment (jnasselle) [reply src/shared_modules/utils/linuxInfoHelper.h:41]: Current design prefer to be silent about this type of exceptions

**Review by RamiroRD — APPROVED** [review]

**Review by Dwordcito — APPROVED** [review]

**Review by HanesSciarrone — APPROVED** [review]

**Pull request merged**